src/v0/destinations/klaviyo/transform.js

```diff
@@ -275,9 +301,101 @@ const process = async (event) => {
+    if (processedEvent.message.length === 2) {
+      identifyResponseList.push(event.message[0]);
+      [processedEvent.message] = event.message.slice(1);
+    } else {
+      [processedEvent.message] = processedEvent.message;
+    }
+  });
+  const subscribeEventGroups = _.groupBy(
+    successRespList,
+    (event) => event.message.body.JSON.data.attributes.list_id,
+  );
+  Object.keys(subscribeEventGroups).forEach((listId) => {
+    // eventChunks = [[e1,e2,e3,..batchSize],[e1,e2,e3,..batchSize]..]
+    const eventChunks = _.chunk(subscribeEventGroups[listId], MAX_BATCH_SIZE);
+    eventChunks.forEach((chunk) => {
+      const batchEventResponse = generateBatchedPaylaodForArray(chunk);
+      batchedResponseList.push(
+        getSuccessRespEvents(
+          batchEventResponse.batchedRequest,
+          batchEventResponse.metadata,
+          batchEventResponse.destination,
+          true,
+        ),
+      );
+    });
+  });
+  identifyResponseList.forEach((response) => {
+    batchedResponseList[0].batchedRequest.push(response);
+  });
+  return batchedResponseList;
+};
+
+// This function separates subscribe response and other responses in chunks
+const getEventChunks = (event, subscribeRespList, otherRespList) => {
+  if (Array.isArray(event.message)) {
+    subscribeRespList.push(event);
+  } else {
+    otherRespList.push(event);
+  }
+};
+
 const processRouterDest = async (inputs, reqMetadata) => {
-  const respList = await simpleProcessRouterDest(inputs, process, reqMetadata);
-  return respList;
+  const errorRespEvents = checkInvalidRtTfEvents(inputs);
+  if (errorRespEvents.length > 0) {
+    return errorRespEvents;
+  }
+  let batchResponseList = [];
+  const batchErrorRespList = [];
+  const subscribeRespList = [];
+  const otherRespList = [];
+  const { destination } = inputs[0];
+  await Promise.all(
+    inputs.map(async (event) => {
+      try {
+        if (event.message.statusCode) {
+          // already transformed event
+          getEventChunks(
+            { message: event.message, metadata: event.metadata, destination },
+            subscribeRespList,
+            otherRespList,
+          );
+        } else {
+          // if not transformed
+          getEventChunks(
+            {
+              message: await process(event),
+              metadata: event.metadata,
+              destination,
+            },
+            subscribeRespList,
+            otherRespList,
+          );
+        }
+      } catch (error) {
+        const errRespEvent = handleRtTfSingleEventError(event, error, reqMetadata);
+        batchErrorRespList.push(errRespEvent);
+      }
+    }),
+  );
+  let batchedSubscribeResponseList = [];
+  if (subscribeRespList.length > 0) {
+    batchedSubscribeResponseList = batchEvents(subscribeRespList);
+  }
+  const otherResponseList = [];
+  otherRespList.forEach((resp) => {
+    otherResponseList.push(getSuccessRespEvents(resp.message, [resp.metadata], resp.destination));
```

Comment:
Change to proper names `otherResponseList` and `otherRespList` are very similar doesn't gove context about what they are supposed to contain

Reply:
Also this can be written in the below way
```node.js
const otherResponseList = otherRespList.map(resp => getSuccessRespEvents(resp.message, [resp.metadata], resp.destination))

```
How does `nonSubscribeList` for `otherRespList` & `nonSubscribeSuccessList` for `otherResponseList` sound ?